extras/userdoc/md/documentation/parallel-computing.md

```diff
@@ -81,6 +81,24 @@ spike files and `dat` for analog recordings from the `multimeter`.
 The `label` and `file_extension` of a recording device can be set like any other
 parameter of a node using `SetStatus`.
 
+### Spike exchange and synapse updates
+
+For spikes between neurons, spikes are always exchanged through the global 
+spike exchange mechanism and it is then the thread responsible for the target 
+neuron that updates the synapse and delivers the spike.
+
+For connections involving devices, either spike generators or detectors, 
+things are a bit different. Here, each device is replicated (cloned) on each 
+thread. Thus, spikes from, e.g., a spike_generator, are delivered by the clone
+for a given thread and these spikes are delivered locally, bypassing the gobal 
+exchange mechanism. In this case, pre- and postsynaptic node (sender and 
+target) are handled by the same thread. Similarly, when a neuron emits a spike,
+the spike is locally transmitted to the clone of the spike_detector belonging 
+to the same thread as the neuron.
```

Comment:
"Spike exchange to or from neurons over connections that either originate or terminate at a device (e.g. `spike_generator`->`neuron`, or `neuron`->`spike_detector`) differs in that it bypasses the global spike exchange mechanism. Instead, spikes are delivered thread-locally from or to a replica of the device. In this case, both the pre- and postsynaptic nodes are handled by the thread to which the neuron is assigned to."